pkg/server/server.go

```diff
@@ -1007,7 +1006,7 @@ func (s *Server) resolveTypesystem(ctx context.Context, storeID, modelID string)
 
 	span.SetAttributes(attribute.KeyValue{Key: authorizationModelIDKey, Value: attribute.StringValue(resolvedModelID)})
 	grpc_ctxtags.Extract(ctx).Set(authorizationModelIDKey, resolvedModelID)
-	_ = grpc.SetHeader(ctx, metadata.Pairs(AuthorizationModelIDHeader, resolvedModelID))
+	s.transport.SetHeader(ctx, AuthorizationModelIDHeader, resolvedModelID)
```

Comment:
We already have a test for this change 👍 https://github.com/openfga/openfga/blob/a6e000905c3947355f138fe459ae9c48536823f4/cmd/run/run_test.go#L1149